src/serac/physics/utilities/variational_form/integral.hpp

```diff
@@ -411,9 +521,10 @@ struct Integral {
     using derivative_type = decltype(get_gradient(qf(x_t{}, make_dual(u_du_t{}))));
 
     auto num_quadrature_points = static_cast<uint32_t>(X.Size() / spatial_dim);
-    qf_derivatives.resize(sizeof(derivative_type) * num_quadrature_points);
+    qf_derivatives_.resize(sizeof(derivative_type) * num_quadrature_points);
 
-    auto qf_derivatives_ptr = reinterpret_cast<derivative_type*>(qf_derivatives.data());
+    // FIXME: Strict aliasing rule?
```

Comment:
`qf_derivatives` may not need to be owned by the `Integral` object, we might be able to just make it a `std::shared_ptr` in the scope of constructor (and change the captures on the lambdas that need it). That way, it could just store the actual derivative type instead of going aliasing a `char *`